www/updateUes.php

```diff
@@ -36,91 +57,217 @@
+
+    $attrsToUpdate = getAttributesToUpdate($attrsFromPerun, $attrsToConversion, $attributesFromIdP);
+    if (empty($attrsToUpdate)) {
+        return false;
+    }
+
+    return updateUserExtSource($adapter, $attrsToUpdate, $userExtSource, $perunUserId);
+}
+
+function getUserExtSource($adapter, $extSourceName, $extLogin)
+{
+    try {
+        return $adapter->getUserExtSource($extSourceName, $extLogin);
+    } catch (SimpleSAML\Module\perun\Exception $ex) {
+        Logger::debug(CLASS_PREFIX . 'Caught exception when fetching user ext source, probably does not exist.');
+        Logger::debug(CLASS_PREFIX . $ex->getMessage());
+
+        return null;
+    }
+}
 
-    $attributesFromPerunRaw = $adapter->getUserExtSourceAttributes($userExtSource['id'], array_keys($attrMap));
+function getAttributesFromPerun($adapter, $attrMap, $userExtSource): array
+{
+    $attributesFromPerunRaw = $adapter->getUserExtSourceAttributes($userExtSource[ID], array_keys($attrMap));
     $attributesFromPerun = [];
+
     foreach ($attributesFromPerunRaw as $attributeFromPerunRaw) {
-        $attributesFromPerun[$attributeFromPerunRaw['name']] = $attributeFromPerunRaw;
+        $attributesFromPerun[$attributeFromPerunRaw[NAME]] = $attributeFromPerunRaw;
     }
 
     if (null === $attributesFromPerun) {
-        throw new Exception('perun/www/updateUes.php: getting attributes was not successful.');
+        throw new Exception(CLASS_PREFIX . 'Getting attributes was not successful.');
     }
 
+    return $attributesFromPerun;
+}
+
+function getAttributesToUpdate($attributesFromPerun, $attrsToConversion, $attributesFromIdP): array
+{
     $attributesToUpdate = [];
 
     foreach ($attributesFromPerun as $attribute) {
-        $attrName = $attribute['name'];
+        $attrName = $attribute[NAME];
 
         if (isset($attrMap[$attrName], $attributesFromIdP[$attrMap[$attrName]])) {
             $attr = $attributesFromIdP[$attrMap[$attrName]];
 
             if (in_array($attrName, $attrsToConversion, true)) {
                 $arrayAsString = [''];
+
                 foreach ($attr as $value) {
                     $arrayAsString[0] .= $value . ';';
                 }
+
                 if (!empty($arrayAsString[0])) {
                     $arrayAsString[0] = substr($arrayAsString[0], 0, -1);
                 }
+
                 $attr = $arrayAsString;
             }
 
-            if (strpos($attribute['type'], 'String') ||
-                strpos($attribute['type'], 'Integer') ||
-                strpos($attribute['type'], 'Boolean')) {
+            if (strpos($attribute[TYPE], STRING_TYPE)
+                || strpos($attribute[TYPE], INTEGER_TYPE)
+                || strpos($attribute[TYPE], BOOLEAN_TYPE)
+            ) {
                 $valueFromIdP = $attr[0];
-            } elseif (strpos($attribute['type'], 'Array') || strpos($attribute['type'], 'Map')) {
+            } elseif (strpos($attribute[TYPE], ARRAY_TYPE) || strpos($attribute[TYPE], MAP_TYPE)) {
```

Comment:
Again, replace the condition with method call with an explanatory name (i.e. `isComplexType`);